### DISCUSSION TITLE
Accounting for trend and seasonality

### DISCUSSION THREAD

**Discussion by shubhambooking2** [google/meridian#178] [General]
I understand the merdian incorporates the idea of trend and seasonality using time varying intercept which is model using knots. however what is not clear to me is how the seasonality is accounted for when we dont have any knots specified i.e. we assume the model to time - invariant intercept

**Comment by brendalewisprice** [comment]
The knots feature allows users to model seasonality using time varying intercepts. So, when the knots = 1, you are correct that the model no longer estimates a time varying intercept, and thus no longer directly models seasonality. However, if there is some seasonal trend in controls, that might actually provide some indirect modeling of seasonality in the baseline, even if it is mixed up in the control estimates.

The default setting for the national model is knots = 1, however we strongly encourage users to consider choosing knots > 1 but also less than n_times. The choice should make sense given the business use case and the amount of data available (weekly vs monthly, number of years). For geo level models, the default is knots = n_times, but users are again encouraged to think about their business use case in the context of the bias-variance tradeoff and consider if it makes sense to use fewer or custom knots.

For a deeper discussion on knots and considerations for setting knots please see our documentation at https://developers.google.com/meridian/docs/advanced-modeling/setting-knots.
